li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/metadataservice/MarioMetadataServiceClient.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.metadataservice;
+
+import com.linkedin.kafka.clients.common.ClusterDescriptor;
+import com.linkedin.kafka.clients.common.ClusterGroupDescriptor;
+import com.linkedin.kafka.clients.utils.LiKafkaClientsUtils;
+import com.linkedin.mario.client.MarioClient;
+import com.linkedin.mario.client.models.v1.TopicQuery;
+import com.linkedin.mario.common.models.v1.KafkaClusterDescriptor;
+import com.linkedin.mario.common.models.v1.KafkaTopicModel;
+import com.linkedin.mario.common.models.v1.TopicQueryResults;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.kafka.common.TopicPartition;
+
+
+// This is a client for Mario metadata service.
+public class MarioMetadataServiceClient implements MetadataServiceClient {
+  private static final int DEFAULT_MAX_RETRIES = 3;
+
+  private final MarioClient _marioClient;
+  private final int _maxRetries;
+
+  // TODO: remove this when Mario supports client registration
+  private Map<UUID, ClusterGroupDescriptor> _clientIdToClusterGroupMap;
+
+  public MarioMetadataServiceClient(String serviceURI) {
+    this(new MarioClient(serviceURI));
+  }
+
+  public MarioMetadataServiceClient(MarioClient marioClient) {
+    this(marioClient, DEFAULT_MAX_RETRIES);
+  }
+
+  public MarioMetadataServiceClient(MarioClient marioClient, int maxRetries) {
+    _maxRetries = maxRetries;
+    _marioClient = marioClient;
+    _clientIdToClusterGroupMap = new HashMap<>();
+  }
+
+  @Override
+  public void configure(Map<String, ?> configs) {
+  }
+
+  @Override
+  public UUID registerFederatedClient(ClusterGroupDescriptor clusterGroup, Map<String, ?> configs, int timeoutMs) {
+    if (clusterGroup == null) {
+      throw new IllegalArgumentException("cluster group cannot be null");
+    }
+
+    // This is a temporary hack until Mario supports client registration.
+    UUID clientId = LiKafkaClientsUtils.randomUUID();
+    _clientIdToClusterGroupMap.put(clientId, clusterGroup);
+    return clientId;
+  }
+
+  @Override
+  public ClusterDescriptor getClusterForTopic(UUID clientId, String topicName, int timeoutMs)
+      throws MetadataServiceClientException {
+    Set<ClusterDescriptor> cluster = getClusterMapForTopics(clientId, new HashSet<>(Arrays.asList(topicName)),
+        timeoutMs).get(topicName);
+    if (cluster == null || cluster.isEmpty()) {
+      return null;
+    }
+    // Unless topic move is in progress, a topic must exist in only one cluster.
+    if (cluster.size() > 1) {
+      throw new IllegalStateException("topic " + topicName + " exists in more than one cluster");
```

Comment:
also print the clusters, and the cluster group arg used

Reply:
done